Import.sql not trimming blank space at the end of an sql statement

**Describe the bug**
Updating from Quarkus 1.71. to Quarkus 1.7.2 **import.sql** file gets an error on startup. In Quarkus 1.7.1 is working, I've tested the import.sql file with SQLIde to the if been a problem of sql file, and line by line.
**I've recognized the bug, Parse is not trimming blank space at the final of the sql statement with ;**

**Expected behavior**
Works in Quarkus 1.7.1. or maybe is a new behaviour.

**Actual behavior**
Not working in Quarkus 1.7.2 . Gets :
2020-09-08 10:12:19,847 ERROR [io.qua.application] (Quarkus Main Thread) Failed to start application (with profile dev): javax.persistence.PersistenceException: [PersistenceUnit: default] Unable to build Hibernate SessionFactory
        at io.quarkus.hibernate.orm.runtime.boot.FastBootEntityManagerFactoryBuilder.persistenceException(FastBootEntityManagerFactoryBuilder.java:115)
        at io.quarkus.hibernate.orm.runtime.boot.FastBootEntityManagerFactoryBuilder.build(FastBootEntityManagerFactoryBuilder.java:75)
        at io.quarkus.hibernate.orm.runtime.FastBootHibernatePersistenceProvider.createEntityManagerFactory(FastBootHibernatePersistenceProvider.java:54)
        at javax.persistence.Persistence.createEntityManagerFactory(Persistence.java:80)
        at javax.persistence.Persistence.createEntityManagerFactory(Persistence.java:55)
        at io.quarkus.hibernate.orm.runtime.JPAConfig$LazyPersistenceUnit.get(JPAConfig.java:163)
        at io.quarkus.hibernate.orm.runtime.JPAConfig.startAll(JPAConfig.java:84)
        at io.quarkus.hibernate.orm.runtime.HibernateOrmRecorder.startAllPersistenceUnits(HibernateOrmRecorder.java:95)
        at io.quarkus.deployment.steps.HibernateOrmProcessor$startPersistenceUnits-1681903524.deploy_0(HibernateOrmProcessor$startPersistenceUnits-1681903524.zig:74)
        at io.quarkus.deployment.steps.HibernateOrmProcessor$startPersistenceUnits-1681903524.deploy(HibernateOrmProcessor$startPersistenceUnits-1681903524.zig:40)
        at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:729)
        at io.quarkus.runtime.Application.start(Application.java:90)
        at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:92)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:61)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:38)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:106)
        at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:29)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at io.quarkus.runner.bootstrap.StartupActionImpl$3.run(StartupActionImpl.java:145)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: org.hibernate.tool.schema.ast.SqlScriptParserException: RecognitionException(@1:2371) : Import script Sql statements must terminate with a ';' char
        at org.hibernate.tool.schema.ast.SqlScriptParser.failIfAnyErrors(SqlScriptParser.java:106)
        at org.hibernate.tool.schema.ast.SqlScriptParser.parseScript(SqlScriptParser.java:59)
        at org.hibernate.tool.schema.ast.SqlScriptParser.extractCommands(SqlScriptParser.java:34)
        at org.hibernate.tool.hbm2ddl.MultipleLinesSqlCommandExtractor.extractCommands(MultipleLinesSqlCommandExtractor.java:24)
        at org.hibernate.tool.schema.internal.exec.AbstractScriptSourceInput.read(AbstractScriptSourceInput.java:40)
        at org.hibernate.tool.schema.internal.SchemaCreatorImpl.applyImportSources(SchemaCreatorImpl.java:491)
        at org.hibernate.tool.schema.internal.SchemaCreatorImpl.performCreation(SchemaCreatorImpl.java:180)
        at org.hibernate.tool.schema.internal.SchemaCreatorImpl.doCreation(SchemaCreatorImpl.java:135)
        at org.hibernate.tool.schema.internal.SchemaCreatorImpl.doCreation(SchemaCreatorImpl.java:121)
        at org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator.performDatabaseAction(SchemaManagementToolCoordinator.java:156)
        at org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator.process(SchemaManagementToolCoordinator.java:73)
        at org.hibernate.internal.SessionFactoryImpl.<init>(SessionFactoryImpl.java:316)
        at io.quarkus.hibernate.orm.runtime.boot.FastBootEntityManagerFactoryBuilder.build(FastBootEntityManagerFactoryBuilder.java:73)
        ... 21 more




**To Reproduce**
Steps to reproduce the behavior:
1. Open import.sql that has some sql statement
2. Intentionally add one or more blank space at the end of an sql statement.
3. And try to run the app, it's get the error mentions above.


**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: `Linux nyj 5.4.0-45-generic #49-Ubuntu SMP Wed Aug 26 13:38:52 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux`  
 - Output of `java -version`:  `openjdk version "11.0.8" 2020-07-14`  
 - Quarkus version or git rev: `1.7.2`  
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): 
`Gradle 6.4.1`  
`Build time:   2020-05-15 19:43:40 UTC`  
`Revision:     1a04183c502614b5c80e3d603074e0b4a2777c5`  
```

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -86,7 +86,7 @@
         <commons-lang3.version>3.9</commons-lang3.version>
         <commons-codec.version>1.14</commons-codec.version>
         <classmate.version>1.3.4</classmate.version>
-        <hibernate-orm.version>5.4.22.Final</hibernate-orm.version>
+        <hibernate-orm.version>5.4.23.Final</hibernate-orm.version>
         <hibernate-reactive.version>1.0.0.Alpha10</hibernate-reactive.version>
         <hibernate-validator.version>6.1.6.Final</hibernate-validator.version>
         <hibernate-search.version>6.0.0.Beta11</hibernate-search.version>

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -87,7 +87,7 @@
         <commons-codec.version>1.14</commons-codec.version>
         <classmate.version>1.3.4</classmate.version>
         <hibernate-orm.version>5.4.23.Final</hibernate-orm.version>
-        <hibernate-reactive.version>1.0.0.Alpha10</hibernate-reactive.version>
+        <hibernate-reactive.version>1.0.0.Alpha11</hibernate-reactive.version>
         <hibernate-validator.version>6.1.6.Final</hibernate-validator.version>
         <hibernate-search.version>6.0.0.Beta11</hibernate-search.version>
         <narayana.version>5.10.6.Final</narayana.version>

--- a/extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateEntityEnhancer.java
+++ b/extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateEntityEnhancer.java
@@ -11,7 +11,7 @@ import org.objectweb.asm.ClassVisitor;
 import org.objectweb.asm.ClassWriter;
 
 import io.quarkus.deployment.QuarkusClassWriter;
-import net.bytebuddy.utility.OpenedClassReader;
+import io.quarkus.gizmo.Gizmo;
 
 /**
  * Used to transform bytecode by registering to
@@ -41,7 +41,9 @@ public final class HibernateEntityEnhancer implements BiFunction<String, ClassVi
         private final Enhancer enhancer;
 
         public HibernateEnhancingClassVisitor(String className, ClassVisitor outputClassVisitor) {
-            super(OpenedClassReader.ASM_API, new QuarkusClassWriter(ClassWriter.COMPUTE_FRAMES | ClassWriter.COMPUTE_MAXS));
+            //Careful: the ASM API version needs to match the ASM version of Gizmo, not the one from Byte Buddy.
+            //Most often these match - but occasionally they will diverge which is acceptable as Byte Buddy is shading ASM.
+            super(Gizmo.ASM_API_VERSION, new QuarkusClassWriter(ClassWriter.COMPUTE_FRAMES | ClassWriter.COMPUTE_MAXS));
             this.className = className;
             this.outputClassVisitor = outputClassVisitor;
             //note that as getLoadingClassLoader is resolved immediately this can't be created until transform time

--- a/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/FastBootHibernatePersistenceProvider.java
+++ b/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/FastBootHibernatePersistenceProvider.java
@@ -153,7 +153,7 @@ public final class FastBootHibernatePersistenceProvider implements PersistencePr
                 continue;
             }
 
-            RecordedState recordedState = PersistenceUnitsHolder.getRecordedState(persistenceUnitName);
+            RecordedState recordedState = PersistenceUnitsHolder.popRecordedState(persistenceUnitName);
 
             if (recordedState.isReactive()) {
                 throw new IllegalStateException(

--- a/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/PersistenceUnitsHolder.java
+++ b/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/PersistenceUnitsHolder.java
@@ -48,13 +48,13 @@ public final class PersistenceUnitsHolder {
         return persistenceUnits.units;
     }
 
-    public static RecordedState getRecordedState(String persistenceUnitName) {
+    public static RecordedState popRecordedState(String persistenceUnitName) {
         checkJPAInitialization();
         Object key = persistenceUnitName;
         if (persistenceUnitName == null) {
             key = NO_NAME_TOKEN;
         }
-        return persistenceUnits.recordedStates.get(key);
+        return persistenceUnits.recordedStates.remove(key);
     }
 
     private static List<PersistenceUnitDescriptor> convertPersistenceUnits(

--- a/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/recording/PrevalidatedQuarkusMetadata.java
+++ b/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/recording/PrevalidatedQuarkusMetadata.java
@@ -49,6 +49,7 @@ public final class PrevalidatedQuarkusMetadata implements Metadata {
 
     public static PrevalidatedQuarkusMetadata validateAndWrap(final MetadataImpl original) {
         original.validate();
+        original.getBootstrapContext().getReflectionManager().reset();
         return new PrevalidatedQuarkusMetadata(original);
     }
 

--- a/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/service/FlatClassLoaderService.java
+++ b/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/service/FlatClassLoaderService.java
@@ -99,6 +99,20 @@ public class FlatClassLoaderService implements ClassLoaderService {
     }
 
     @Override
+    public Package packageForNameOrNull(String packageName) {
+        try {
+            Class<?> aClass = Class.forName(packageName + ".package-info", false, getClassLoader());
+            return aClass == null ? null : aClass.getPackage();
+        } catch (ClassNotFoundException e) {
+            log.packageNotFound(packageName);
+            return null;
+        } catch (LinkageError e) {
+            log.warn("LinkageError while attempting to load Package named " + packageName, e);
+            return null;
+        }
+    }
+
+    @Override
     public <T> T workWithClassLoader(Work<T> work) {
         ClassLoader systemClassLoader = getClassLoader();
         return work.doWork(systemClassLoader);

--- a/extensions/hibernate-reactive/runtime/src/main/java/io/quarkus/hibernate/reactive/runtime/FastBootHibernateReactivePersistenceProvider.java
+++ b/extensions/hibernate-reactive/runtime/src/main/java/io/quarkus/hibernate/reactive/runtime/FastBootHibernateReactivePersistenceProvider.java
@@ -123,7 +123,7 @@ public final class FastBootHibernateReactivePersistenceProvider implements Persi
                 continue;
             }
 
-            RecordedState recordedState = PersistenceUnitsHolder.getRecordedState(persistenceUnitName);
+            RecordedState recordedState = PersistenceUnitsHolder.popRecordedState(persistenceUnitName);
 
             final PrevalidatedQuarkusMetadata metadata = recordedState.getMetadata();
             final BuildTimeSettings buildTimeSettings = recordedState.getBuildTimeSettings();

